src/sidebar/util/test/is-third-party-service-test.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const proxyquire = require('proxyquire');
+
+describe('sidebar.util.isThirdPartyService', () => {
+  let fakeServiceConfig;
+  let fakeSettings;
+  let isThirdPartyService;
+
+  beforeEach(() => {
+    fakeServiceConfig = sinon.stub();
+    fakeSettings = {authDomain: 'hypothes.is'};
+
+    isThirdPartyService = proxyquire('../is-third-party-service', {
```

Comment:
This works, but you could avoid the mocking here by requiring the argument to be the configured service, or just passing some representative settings and not mocking `service-config`.

Reply:
If `isThirdPartyService()` required the individual service object then every time anyone wanted to call it they'd have to do:

```javascript
const serviceConfig = require('../service-config');
var service = serviceConfig(settings);
if (isThirdPartyService(settings, service)) { ... };
```

Instead of just:

```javascript
if (isThirdPartyService(settings)) { ... };
```

This may mean that the tests for `isThirdPartyService()` no longer need to mock `serviceConfig()`, but the tests for everything that calls `isThirdPartyService()` would need to mock `serviceConfig()` instead.

Or to put it another way - we wouldn't be encapsulating `serviceConfig()`. If you were to draw a diagram of the architecture, instead of this:

![1](https://user-images.githubusercontent.com/22498/33672033-1890e886-daa1-11e7-9372-78bdd090e984.JPG)

(Here I've used `AnnotationService` as a stand-in for a new class that includes `isThirdPartyService()` and other service-related methods in one class)

You would instead have this:

![2](https://user-images.githubusercontent.com/22498/33672060-2841fff4-daa1-11e7-84ff-ef70da00d394.JPG)